app/api/bucketlist_api.py

```diff
@@ -0,0 +1,67 @@
+from flask import g, request, jsonify, abort
+from app import app, db
+from app.auth.auth import auth
+from flask_login import login_required
+from app.models import Bucketlist
+
+@app.route('/bucketlists/', methods = ['GET'])
+@login_required
+def get_all_bucketlists(self):
+    """gets all the bucket lists"""
+    bucket = Bucketlist.query.filter_by(created_by=g.user_id)
+    if not bucket or bucket.created_by != g.user_id:
+        return jsonify(message='There is No Bucketlist to display!!!!')
+    else:
+        return bucket, 200
+
+@app.route('/bucketlists/', methods = ['POST'])
+@login_required
+def new_bucketlist():
+    """Create a new bucket list"""
+    if not request.json:
+        abort(400)
+    bucketlist = Bucketlist(title=request.get.json('title'), created_by=g.user_id)
+    db.session.add(bucketlist)
```

Comment:
You can consider having a save method in your `bucketlist` model to avoid constantly doing `db.session.add(bucketlist)`.

Something like this in your models:
```
def save(self):
        """
        Save to database.
        Save instance of the object to database and commit.
        """
        db.session.add(self)
        db.session.commit()
``` for saving to the database and 
```
 def delete(self):
        """
        Delete from database.
        Deletes instance of an object from database
        """
        db.session.delete(self)
        db.session.commit()
``` for deleting from the database.